src/Interceptor/EventInterceptor.php

```diff
@@ -0,0 +1,114 @@
+
+use OpenClassrooms\ServiceProxy\Annotation\Event;
+use OpenClassrooms\ServiceProxy\Annotation\Exception\InvalidEventNameException;
+use OpenClassrooms\ServiceProxy\Contract\EventHandler;
+use OpenClassrooms\ServiceProxy\Interceptor\Contract\PrefixInterceptor;
+use OpenClassrooms\ServiceProxy\Interceptor\Contract\SuffixInterceptor;
+use OpenClassrooms\ServiceProxy\Interceptor\Request\Instance;
+use OpenClassrooms\ServiceProxy\Interceptor\Response\Response;
+
+final class EventInterceptor extends AbstractInterceptor implements SuffixInterceptor, PrefixInterceptor
+{
+    protected int $prefixPriority = 20;
+
+    protected int $suffixPriority = 10;
+
+    /**
+     * @throws InvalidEventNameException
+     */
+    public function prefix(Instance $instance): Response
+    {
+        $annotations = $instance->getMethod()
+            ->getAnnotations(Event::class);
+
+        foreach ($annotations as $annotation) {
+            $handler = $this->getHandler(EventHandler::class, $annotation);
+            if ($annotation->hasMethod(Event::PRE_METHOD)) {
+                $event = $handler->make(
+                    $this->getEventName($instance, $annotation, Event::PRE_METHOD),
+                    $instance->getMethod()
+                        ->getParameters()
+                );
+                $handler->send($event);
+            }
+        }
+
+        return new Response();
+    }
+
+    /**
+     * @throws InvalidEventNameException
+     */
+    public function suffix(Instance $instance): Response
+    {
+        $annotations = $instance->getMethod()
+            ->getAnnotations(Event::class);
+
+        foreach ($annotations as $annotation) {
+            $handler = $this->getHandler(EventHandler::class, $annotation);
+
+            if ($annotation->hasMethod(Event::POST_METHOD) && !$instance->getMethod()->threwException()) {
+                $event = $handler->make(
+                    $this->getEventName($instance, $annotation, Event::POST_METHOD),
+                    $instance->getMethod()
+                        ->getParameters(),
+                    $instance->getMethod()
+                        ->getReturnedValue(),
+                );
+                $handler->send($event);
+            }
+
+            if ($annotation->hasMethod(Event::ON_EXCEPTION_METHOD) && $instance->getMethod()->threwException()) {
+                $event = $handler->make(
+                    $this->getEventName($instance, $annotation, Event::ON_EXCEPTION_METHOD),
+                    $instance->getMethod()
+                        ->getParameters(),
+                    null,
+                    $instance->getMethod()
+                        ->getException()
+                );
+                $handler->send($event);
+            }
+        }
+
+        return new Response();
+    }
+
+    public function supportsPrefix(Instance $instance): bool
+    {
+        return $instance->getMethod()
+            ->hasAnnotation(Event::class);
```

Comment:
```suggestion
        return $this->supportsSuffix($instance);
```

Reply:
or in reverse, the idea is to keep the consistency with other interceptors, what do you think ?